d2core/d2map/d2maprenderer/tile_cache.go

```diff
@@ -40,22 +40,22 @@ func (mr *MapRenderer) generateTileCache() {
 
 func (mr *MapRenderer) generateFloorCache(tile *d2ds1.FloorShadowRecord, tileX, tileY int) {
 	tileOptions := mr.mapEngine.GetTiles(int32(tile.Style), int32(tile.Sequence), 0)
-	var tileData []*d2dt1.Tile
+	var tileData []d2dt1.Tile
 	var tileIndex byte
 
 	if tileOptions == nil {
 		log.Printf("Could not locate tile Style:%d, Seq: %d, Type: %d\n", tile.Style, tile.Sequence, 0)
-		tileData = append(tileData, &d2dt1.Tile{})
+		tileData = append(tileData, d2dt1.Tile{})
 		tileData[0].Width = 10
 		tileData[0].Height = 10
 	} else {
 		if !tileOptions[0].MaterialFlags.Lava {
 			tileIndex = mr.getRandomTile(tileOptions, tileX, tileY, mr.mapEngine.Seed())
-			tileData = append(tileData, &tileOptions[tileIndex])
+			tileData = append(tileData, tileOptions[tileIndex])
```

Comment:
Maybe I'm over-suspicious because of the assignment thing, but doesn't this copy the like option at index "tileIndex"? Since it's now a slice of struct instead of *struct?